failed to install

git clone https://github.com/GoogleChrome/rendertron.git
cd rendertron
npm install
npm run build

I used this commond to install. But It failded:
_src/rendertron.ts:146:14 - error TS2769: No overload matches this call.
  The last overload gave the following error.
    Argument of type '"unhandledRejection"' is not assignable to parameter of type 'Signals'.

146   process.on('unhandledRejection', logUncaughtError);
                 ~~~~~~~~~~~~~~~~~~~~

  node_modules/@types/node/globals.d.ts:1012:9
    1012         on(event: Signals, listener: SignalsListener): this;
                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    The last overload is declared here.


Found 1 error.


npm ERR! Linux 4.15.0-52-generic
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "run" "build"
npm ERR! node v8.10.0
npm ERR! npm  v3.5.2
npm ERR! code ELIFECYCLE
npm ERR! rendertron@2.0.1 build: `tsc`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the rendertron@2.0.1 build script 'tsc'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the rendertron package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     tsc
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs rendertron
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls rendertron
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /root/js_load/rendertron/npm-debug.log_

--- a/package.json
+++ b/package.json
@@ -32,10 +32,10 @@
     "koa": "^2.5.2",
     "koa-bodyparser": "^4.2.1",
     "koa-compress": "^3.0.0",
+    "koa-logger": "^3.2.0",
     "koa-route": "^3.2.0",
     "koa-send": "^5.0.0",
     "koa-static": "^5.0.0",
-    "koa-logger": "^3.2.0",
     "performance-now": "^2.1.0",
     "progress-bar-element": "^2.0.1",
     "puppeteer": "^1.7.0",
@@ -51,12 +51,13 @@
     "@types/koa-route": "^3.2.4",
     "@types/koa-send": "^4.1.1",
     "@types/koa-static": "^4.0.0",
+    "@types/node": "^12.7.4",
     "@types/puppeteer": "^1.6.0",
     "@types/supertest": "^2.0.4",
     "ava": "1.0.0-beta.6",
     "nodemon": "^1.18.3",
     "supertest": "^3.0.0",
     "tslint": "^5.11.0",
-    "typescript": "^3.0.1"
+    "typescript": "^3.6.2"
   }
 }

--- a/src/rendertron.ts
+++ b/src/rendertron.ts
@@ -152,7 +152,7 @@ async function logUncaughtError(error: Error) {
 // The type for the unhandleRejection handler is set to contain Promise<any>,
 // so we disable that linter rule for the next line
 // tslint:disable-next-line: no-any
-async function logUnhandledRejection(reason: {} | null | undefined, _: Promise<any>) {
+async function logUnhandledRejection(reason: unknown, _: Promise<any>) {
   console.error('Unhandled rejection');
   console.error(reason);
   process.exit(1);

--- a/src/rendertron.ts
+++ b/src/rendertron.ts
@@ -8,8 +8,8 @@ import * as path from 'path';
 import * as puppeteer from 'puppeteer';
 import * as url from 'url';
 
-import {Renderer, ScreenshotError} from './renderer';
-import {Config, ConfigManager} from './config';
+import { Renderer, ScreenshotError } from './renderer';
+import { Config, ConfigManager } from './config';
 
 /**
  * Rendertron rendering service. This runs the server which routes rendering
@@ -18,7 +18,7 @@ import {Config, ConfigManager} from './config';
 export class Rendertron {
   app: Koa = new Koa();
   private config: Config = ConfigManager.config;
-  private renderer: Renderer|undefined;
+  private renderer: Renderer | undefined;
   private port = process.env.PORT || this.config.port;
 
   async initialize() {
@@ -27,7 +27,7 @@ export class Rendertron {
 
     this.port = this.port || this.config.port;
 
-    const browser = await puppeteer.launch({args: ['--no-sandbox']});
+    const browser = await puppeteer.launch({ args: ['--no-sandbox'] });
     this.renderer = new Renderer(browser, this.config);
 
     this.app.use(koaLogger());
@@ -38,17 +38,17 @@ export class Rendertron {
 
     this.app.use(route.get('/', async (ctx: Koa.Context) => {
       await koaSend(
-        ctx, 'index.html', {root: path.resolve(__dirname, '../src')});
+        ctx, 'index.html', { root: path.resolve(__dirname, '../src') });
     }));
     this.app.use(
       route.get('/_ah/health', (ctx: Koa.Context) => ctx.body = 'OK'));
 
     // Optionally enable cache for rendering requests.
     if (this.config.cache === 'datastore') {
-      const {DatastoreCache} = await import('./datastore-cache');
+      const { DatastoreCache } = await import('./datastore-cache');
       this.app.use(new DatastoreCache().middleware());
     } else if (this.config.cache === 'memory') {
-      const {MemoryCache} = await import('./memory-cache');
+      const { MemoryCache } = await import('./memory-cache');
       this.app.use(new MemoryCache().middleware());
     }
 
@@ -127,7 +127,7 @@ export class Rendertron {
 
     try {
       const img = await this.renderer.screenshot(
-          url, mobileVersion, dimensions, options);
+        url, mobileVersion, dimensions, options);
 
       for (const key in this.config.headers) {
         ctx.set(key, this.config.headers[key]);
@@ -149,11 +149,17 @@ async function logUncaughtError(error: Error) {
   process.exit(1);
 }
 
+async function logUnhandledRejection(reason: {} | null | undefined, _: Promise<any>) {
+  console.error('Unhandled rejection');
+  console.error(reason);
+  process.exit(1);
+}
+
 // Start rendertron if not running inside tests.
 if (!module.parent) {
   const rendertron = new Rendertron();
   rendertron.initialize();
 
   process.on('uncaughtException', logUncaughtError);
-  process.on('unhandledRejection', logUncaughtError);
+  process.on('unhandledRejection', logUnhandledRejection);
 }

--- a/src/rendertron.ts
+++ b/src/rendertron.ts
@@ -149,6 +149,9 @@ async function logUncaughtError(error: Error) {
   process.exit(1);
 }
 
+// The type for the unhandleRejection handler is set to contain Promise<any>,
+// so we disable that linter rule for the next line
+// tslint:disable-next-line: no-any
 async function logUnhandledRejection(reason: {} | null | undefined, _: Promise<any>) {
   console.error('Unhandled rejection');
   console.error(reason);

